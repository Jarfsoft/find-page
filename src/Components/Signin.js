import { React, useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useHistory } from 'react-router';

import { getUser } from '../userApi';
import actions from '../Actions';
import './Signin.css';

function Signin() {
  const [name, setName] = useState();
  const user = useSelector((state) => state.users.userId);
  const dispatch = useDispatch();
  const setUser = (n) => {
    dispatch(actions.login(n));
  };
  const submitHandler = (e) => {
    e.preventDefault();
    getUser(name, setUser);
  };
  const history = useHistory();

  useEffect(() => {
    if (user !== 0) {
      history.push('/');
    }
  }, [user]);

  const changeHandler = (e) => {
    setName(e.target.value);
  };
  return (
    <form onSubmit={submitHandler}>
      <h1>Name:</h1>
      <br />
      <input id="name" required type="text" name="name" value={name} onChange={changeHandler} />
      <br />
      <input className="submit-btn" type="submit" value="Sign in" />
    </form>
  );
}

export default Signin;
