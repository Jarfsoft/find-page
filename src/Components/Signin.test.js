import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import Signin from './Signin';

describe('Navbar render', () => {
  const mockStore = configureStore();
  let store;
  const state = { 
    users: {
      userId: 0,
      name: '',
    },
  };

  describe('Snapshot check', () => {
    it('should render Signin component', () => {
      store = mockStore(state);
      const signin = renderer
        .create(
          <Provider store={store}>
            <BrowserRouter>
              <Signin />
            </BrowserRouter>
          </Provider>
        ).toJSON();
      expect(signin).toMatchSnapshot();
    });
  });
  it('should containt a form', () => {
    store = mockStore(state);
    const signin = render(
      <BrowserRouter>
        <Provider store={store}>
          <Signin />
        </Provider>
      </BrowserRouter>,
    );
    const form = signin.container.querySelectorAll('form');
    expect(form).toHaveLength(1);
  });
});