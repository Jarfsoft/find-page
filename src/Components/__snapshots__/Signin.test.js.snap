// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar render Snapshot check should render Signin component 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Name:
  </h1>
  <br />
  <input
    id="name"
    name="name"
    onChange={[Function]}
    required={true}
    type="text"
  />
  <br />
  <input
    className="submit-btn"
    type="submit"
    value="Sign in"
  />
  <br />
</form>
`;
